pkg/operator/ceph/csi/spec.go

```diff
@@ -935,3 +957,76 @@ func GenerateNetNamespaceFilePath(ctx context.Context, client client.Client, clu
 func generateNetNamespaceFilePath(kubeletDirPath, driverFullName, clusterNamespace string) string {
 	return fmt.Sprintf("%s/plugins/%s/%s.net.ns", kubeletDirPath, driverFullName, clusterNamespace)
 }
+
+func validateCSIDriverNamePrefix(ctx context.Context, clientset kubernetes.Interface, namespace, driverNamePrefix string) error {
+	if EnableRBD {
+		rbdDriverNamePrefix, err := getCSIDriverNamePrefixFromDeployment(ctx, clientset, namespace, csiRBDProvisioner, "csi-rbdplugin")
+		if err != nil {
+			return err
+		}
+		if rbdDriverNamePrefix != "" && rbdDriverNamePrefix != driverNamePrefix {
+			return errors.Errorf("rbd driver already exists with prefix %q, cannot use prefix %q", rbdDriverNamePrefix, driverNamePrefix)
+		}
+		return nil
```

Comment:
If the driver was not found, looks like this will return here instead of checking the next driver. If no prefix was found, shall we check the next driver?

Reply:
i missed that case, move the check as required, this should cover both cases now. Thanks